File: BOJ/boj1149.cpp
// 1149 - RGB°Å¸® 
#include <stdio.h>
#include <algorithm>

using namespace std;

int dp[1001][3];
int N;
int r,g,b;

int main(void) {
	
	scanf("%d",&N);
	for(int i = 0; i < N; i++) {
		scanf("%d %d %d",&r ,&g ,&b);
		dp[i][0] = r;
		dp[i][1] = g;
		dp[i][2] = b;
	}
	
	for(int i = 0; i < N; i++) {
		dp[i+1][0] = dp[i+1][0] + min(dp[i][1], dp[i][2]);
		dp[i+1][1] = dp[i+1][1] + min(dp[i][0], dp[i][2]);
		dp[i+1][2] = dp[i+1][2] + min(dp[i][0], dp[i][1]);
	}
	
	int tmp = min(dp[N-1][1],dp[N-1][2]);
	printf("%d",min(dp[N-1][0],tmp));
	
	return 0;
}

File: BOJ/boj1149.java
import java.io.*;
import java.util.*;

/*
 * 1149 RGB거리
 */
public class boj1149 {
    static StringBuffer sb = new StringBuffer();
    static int N;
    static int[][] rgb;
    public static void main(String[] args) throws Exception {
        init();
        //TO-DO
        solve();
    }

    public static void init() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Scanner sc = new Scanner(System.in);
        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        //TO-DO
        rgb = new int[N+1][N+1];
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            int j = 0;
            while (st.hasMoreTokens()) {
                rgb[i][j++] = Integer.parseInt(st.nextToken());
            }
        }
        //END
        sc.close();
		br.close();
    }

    public static void solve() {
        for (int i = 0; i < N; i++) {
            rgb[i+1][0] = rgb[i+1][0] + Math.min(rgb[i][1], rgb[i][2]);
            rgb[i+1][1] = rgb[i+1][1] + Math.min(rgb[i][0], rgb[i][2]);
            rgb[i+1][2] = rgb[i+1][2] + Math.min(rgb[i][0], rgb[i][1]);
        }
        int res = Math.min(rgb[N-1][0], Math.min(rgb[N-1][1], rgb[N-1][2]));
        System.out.println(res);
    }
}
